src/Spain/Mews.Fiscalizations.Spain/Model/Request/ForeignCompany.cs

```diff
@@ -1,23 +1,21 @@
-﻿using Mews.Fiscalizations.Core.Model;
-
-namespace Mews.Fiscalizations.Spain.Model.Request
+﻿namespace Mews.Fiscalizations.Spain.Model.Request
 {
-    public sealed class ForeignCompany
+    public class ForeignCompany
     {
-        public ForeignCompany(Name name, ResidenceCountryIdentificatorType identificatiorType, String1To20 id, Country country)
+        public ForeignCompany(
+            Name name,
+            ForeignCompayTaxpayerNumber taxpayerNumber,
+            ResidenceCountryIdentificatorType identificatiorType = ResidenceCountryIdentificatorType.NotSelected)
```

Comment:
I don't understand how this works. So you're saying it needs to be a TaxPayerNumber and you're also saying that there's an enum that tells you what it is. 

So how exactly can I provide a value of `ResidenceCertificate` when I can't provide the `certificateNumber` because it will not pass the validation for ForeignTaxPayerNumber?